Fixing test warnings

dtulga opened the pull request:
This fixes a variety of warnings that show up when running tests:

- Uses `selected_columns` instead of `columns` in select statements to avoid this SQLAlchemy warning: `  /home/runner/work/datachain/datachain/.nox/tests-3-9/lib/python3.9/site-packages/datachain/query/dataset.py:1255: SADeprecationWarning: The SelectBase.c and SelectBase.columns attributes are deprecated and will be removed in a future release; these attributes implicitly create a subquery that should be explicit.  Please call SelectBase.subquery() first in order to create a subquery, which then contains this attribute.  To access the columns that this SELECT object SELECTs from, use the SelectBase.selected_columns attribute. (deprecated since: 1.4)`
- Renames Test... classes to no longer start with Test as this causes pytest to attempt to collect them as tests, and show warnings like: `/home/runner/work/datachain/datachain/tests/unit/lib/test_datachain_merge.py:12: PytestCollectionWarning: cannot collect test class 'TestUser' because it has a __init__ constructor (from: tests/unit/lib/test_datachain_merge.py)`
- Uses `model_dump` instead of `dict` for Pydantic models, because of this warning: `/home/runner/work/datachain/datachain/.nox/tests-3-9/lib/python3.9/site-packages/pydantic/main.py:1087: PydanticDeprecatedSince20: The "dict" method is deprecated; use "model_dump" instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.8/migration/`
- Sort the returned dataframe from `to_pandas` to avoid this warning from pandas: `/home/runner/work/datachain/datachain/tests/func/test_feature_pickling.py:90: PerformanceWarning: indexing past lexsort depth may impact performance.`
- Pytest mocks don't need to be used as a context manager, as indicated by this warning: `/home/runner/work/datachain/datachain/tests/func/test_datasets.py:213: PytestMockWarning: Mocks returned by pytest-mock do not need to be used as context managers. The mocker fixture automatically undoes mocking at the end of a test. This warning can be ignored if it was triggered by mocking a context manager. https://pytest-mock.readthedocs.io/en/latest/remarks.html#usage-as-context-manager`

skshetry commented:
@dtulga, please add a filterwarnings to raise warnings as an error, so we don't introduce them again.

Example:

https://github.com/iterative/dvc/blob/8f50f2509f31f2699408f32658464ff20d20b2ab/pyproject.toml#L164-L166


See https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#controlling-warnings.

dtulga commented on src/datachain/lib/dc.py:
```diff
@@ -1027,7 +1027,11 @@ def to_pandas(self, flatten=False) -> "pd.DataFrame":
+        # Sort the dataframe to avoid a PerformanceWarning about unsorted indexing.
+        df.sort_index(axis=0, inplace=True, sort_remaining=True)
+        df.sort_index(axis=1, inplace=True, sort_remaining=True)
+        return df
```
I moved this sorting to only run for the test where this warning is generated. Having unsorted dataframes can lead to strange performance problems, such as O(N) instead of O(1) operations, so this may end up being a problem for users in the future.

dtulga commented on tests/unit/lib/test_datachain_merge.py:
```diff
@@ -9,38 +9,38 @@
 
 
-class TestUser(BaseModel):
+class ExampleUser(BaseModel):
```
Sure, removed the Example prefix.

dtulga commented:
I have updated this PR to include filterwarnings to convert these fixed warnings into errors, and added code to ignore another warning (that only happens due to strange test code).  Two of the remaining warnings are in third-party code (as I can't find any references to `timm` in the codebase, and the other warning is generated within `botocore`), and they raise just a plain `DeprecationWarning` which I don't want to ignore everywhere, as it may cause us to miss other important deprecations. And the torch warnings seem to have some complexity to them, and I'm not sure what the correct fix is, as I don't have enough experience with PyTorch to determine if they will affect floating-point precision, or cause other problems.

skshetry commented:
cc @dberenbaum, maybe you can help to fix these deprecation warnings (no rush though)?

```
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48
  /home/runner/work/datachain/datachain/.nox/tests-3-9/lib/python3.9/site-packages/timm/models/layers/__init__.py:48: DeprecationWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
    warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", DeprecationWarning)

tests/unit/lib/test_clip.py: 24 warnings
tests/unit/lib/test_text.py: 1 warning
tests/func/test_pytorch.py: 2 warnings
  /home/runner/work/datachain/datachain/.nox/tests-3-9/lib/python3.9/site-packages/datachain/lib/text.py:34: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    tokens = torch.tensor(tokens)

tests/unit/lib/test_clip.py::test_similarity_scores_hf
tests/unit/lib/test_image.py::test_convert_image_hf
  /home/runner/work/datachain/datachain/.nox/tests-3-9/lib/python3.9/site-packages/datachain/lib/image.py:35: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    img = torch.tensor(img.pixel_values[0])  # type: ignore[assignment,attr-defined]

tests/unit/lib/test_clip.py::test_similarity_scores_hf
  /home/runner/work/datachain/datachain/.nox/tests-3-9/lib/python3.9/site-packages/datachain/lib/clip.py:21: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    return lambda x: method(torch.tensor(x))

tests/func/test_pytorch.py::test_pytorch_dataset
tests/func/test_pytorch.py::test_pytorch_dataset_sample
tests/func/test_pytorch.py::test_to_pytorch
  /home/runner/work/datachain/datachain/.nox/tests-3-9/lib/python3.9/site-packages/torchvision/transforms/v2/_deprecated.py:42: UserWarning: The transform `ToTensor()` is deprecated and will be removed in a future release. Instead, please use `v2.Compose([v2.ToImage(), v2.ToDtype(torch.float32, scale=True)])`.Output is equivalent up to float precision.
    warnings.warn(
```
